doc/adrs/217_ensure_elasticsearch_query_results_contain_enough_information.adoc

```diff
@@ -0,0 +1,71 @@
+= ADR-217 - Ensure Elasticsearch query results contain enough information
+
+== Context
+
+Sirius Web can populate Elasticsearch indices with information from project's semantic data.
+
+The main goal of these indices is to provide an API to retrieve information from semantic data without loading the corresponding editing context, for example, one can query Elasticsearch to retrieve all the elements matching a given string, regardless of where they are actually stored.
+
+The minimal information to display an object in Sirius Web is usually composed of a label and an icon.
+This information is computed by `ILabelService` on a semantic object, and can be extended by downstream applications.
+Since we don't want to load editing contexts, and we can't make any assumption about how labels and icons are computed, we need to ensure the Elasticsearch query API returns enough information to display query results.
+
+We should also ensure that client code can load and manipulate the corresponding semantic object if needed (e.g. to display additional information not stored in the index).
+
+== Decision
+
+At the moment, indexed documents contain only the raw data of their source objects.
+As explained above, this information is not sufficient to compute relevant label/icons without loading the source object.
+We will add `label` and `iconURLs` fields to indexed documents to ensure they can be presented to the end user in contexts where the actual semantic object is not accessible.
+
+We will also add an `editingContextId` field to indexed documents to ensure the semantic object can be loaded if needed.
+
+This information will be set during the indexing process (where the semantic object is accessible) using `ILabelService` (for `label` and `iconURL`) and `IEditingContext` (for `editingContextId`).
+
+We will define the following interface, which will contain enough information to display index query results to the end user, as well as load the actual semantic object if needed.
+This interface may evolve in the future to match new requirements like complex ES|QL query results.
+[source, java]
+----
+public interface IndexHit {
```

Comment:
I'm not a fan of `IndexHit`, it seems more like an action related to caching with hit and miss. I would rather use something like `IndexEntry`. Wether we "hit" it or not is another thing.